matcher/cache/elem_map_gen.go

```diff
@@ -0,0 +1,255 @@
+// This file was automatically generated by genny.
+// Any changes will be lost if this file is regenerated.
+// see https://github.com/mauricelam/genny
+
+package cache
+
+// This file was automatically generated by genny.
```

Comment:
Hm why is this showing up twice? Also it'd be ideal if the license is at the top of the file.

Reply:
It's because this is a two level generated file, first one is setting the KeyType but not setting the ValueType (because we have a specialized `ident.ID` keyed map and a specialized `[]byte` keyed map) and the second level of generation is going from the key specialized map and setting a concrete ValueType.

In the interest of shipping let's put off the license issue and fix that asynchronously.